pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -39,12 +39,7 @@ type OpenShiftClusterDynamicValidator interface {
 }
 
 // NewOpenShiftClusterDynamicValidator creates a new OpenShiftClusterDynamicValidator
-func NewOpenShiftClusterDynamicValidator(log *logrus.Entry, env env.Interface, oc *api.OpenShiftCluster, subscriptionDoc *api.SubscriptionDocument) (OpenShiftClusterDynamicValidator, error) {
-	fpAuthorizer, err := env.FPAuthorizer(subscriptionDoc.Subscription.Properties.TenantID, env.Environment().ResourceManagerEndpoint)
-	if err != nil {
-		return nil, err
-	}
-
+func NewOpenShiftClusterDynamicValidator(fpAuthorizer refreshable.Authorizer, log *logrus.Entry, env env.Interface, oc *api.OpenShiftCluster, subscriptionDoc *api.SubscriptionDocument) OpenShiftClusterDynamicValidator {
```

Comment:
As you will be changing can you move `fpAuthorizer` to be after `log` and `env`. At some point, we tried to keep order like: `ctx`, `log`,`env`, `other stuff` for all signatures.